api/nodes/views.py

```diff
@@ -1558,6 +1560,52 @@ def create(self, *args, **kwargs):
         return ret
 
 
+class NodeSubjectsList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin, NodeMixin):
+    """The documentation for this endpoint can be found [here](https://developer.osf.io/#operation/nodes_subjects_list).
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        ContributorOrPublic,
+        ExcludeWithdrawals,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_BASE_READ, CoreScopes.SUBJECTS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    serializer_class = SubjectSerializer
+
+    model = Subject
+    view_category = 'nodes'
+    view_name = 'node-subjects'
+
+    ordering = ('-id',)
+
+    def get_queryset(self):
+        return self.get_node().subjects.all()
+
+
+class NodeSubjectsRelationship(SubjectRelationshipBaseView, NodeMixin):
+    """The documentation for this endpoint can be found [here](https://developer.osf.io/#operation/node_subjects_relationship).
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        ContributorOrPublic,
+        ExcludeWithdrawals,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NODE_BASE_WRITE]
+
+    view_category = 'nodes'
+    view_name = 'node-relationships-subjects'
+
+    ordering = ('-id',)
+
+    def get_resource(self, check_object_permissions=True):
+        return self.get_node(check_object_permissions=check_object_permissions)
```

Comment:
All these ...SubjectsRelationship endpoints - on nodes, registrations, and collection submissions (not preprints), only exist for allowing subjects to be removed entirely.  You can PUT/PATCH these endpoints.  You can either replace all the subjects, or remove all the subjects with an empty list.

I decided against allowing you to issue a POST or DELETE here, because subjects have weird behavior.  If you send a request in with a child subject, the backend will save the whole tree, so save its parent as well.  I didn't want you to be able to delete a single subject or add a single subject.